retrofit2/src/main/java/com/linecorp/armeria/client/retrofit2/ArmeriaCallFactory.java

```diff
@@ -171,12 +174,17 @@ public Request request() {
             return request;
         }
 
-        private synchronized void createRequest() {
-            if (httpResponse != null) {
-                throw new IllegalStateException("executed already");
+        private void createRequest() {
+            reentrantLock.lock();
+            try{
+                if (httpResponse != null) {
+                    throw new IllegalStateException("executed already");
+                }
+                executionStateUpdater.compareAndSet(this, ExecutionState.IDLE, ExecutionState.RUNNING);
+                httpResponse = doCall(callFactory, request);
```

Comment:
It is likely that `httpResponse` can be atomically updated without a lock or `synchronized`.
```java
if (executionStateUpdater.compareAndSet(this, ExecutionState.IDLE, ExecutionState.RUNNING)) {
    httpResponse = doCall(callFactory, request);
} else {
    throw new IllegalStateException("executed already");
}
```

Reply:
so this method should be reverted to previous state then?  just realised the volatile on httpResponse @ikhoon 

> It is likely that `httpResponse` can be atomically updated without a lock or `synchronized`.
> 
> ```java
> if (executionStateUpdater.compareAndSet(this, ExecutionState.IDLE, ExecutionState.RUNNING)) {
>     httpResponse = doCall(callFactory, request);
> } else {
>     throw new IllegalStateException("executed already");
> }
> ```

Reply:
changes reverted.

> It is likely that `httpResponse` can be atomically updated without a lock or `synchronized`.
> 
> ```java
> if (executionStateUpdater.compareAndSet(this, ExecutionState.IDLE, ExecutionState.RUNNING)) {
>     httpResponse = doCall(callFactory, request);
> } else {
>     throw new IllegalStateException("executed already");
> }
> ```

Reply:
> so this method should be reverted to previous state then?

I meant to fix the code to use `if` with `compareAndSet` without `ReentrantLock`.

Reply:
yes, no point calling lock/synchronized on a CAS operation. However this change
`if (executionStateUpdater.compareAndSet(this, ExecutionState.IDLE, ExecutionState.RUNNING)) {
    httpResponse = doCall(callFactory, request);
} else {
    throw new IllegalStateException("executed already");
}`
caused threads not able to update the executionStateUpdater in the ArmeriaCallFactoryTest class to throw the above exception as shown below. Not sure what to do from here.
<img width="1440" alt="Screenshot 2023-01-06 at 23 04 32" src="https://user-images.githubusercontent.com/36740791/211109844-57300f56-1386-4426-9788-83f2adb76c6c.png">